sentry-log4j2/src/main/java/io/sentry/log4j2/SentryAppender.java

```diff
@@ -0,0 +1,220 @@
+            Optional.ofNullable(transport).ifPresent(options::setTransport);
+          });
+    }
+    super.start();
+  }
+
+  @Override
+  public void append(LogEvent eventObject) {
+    if (eventObject.getLevel().isMoreSpecificThan(minimumEventLevel)) {
+      Sentry.captureEvent(createEvent(eventObject));
+    }
+    if (eventObject.getLevel().isMoreSpecificThan(minimumBreadcrumbLevel)) {
+      Sentry.addBreadcrumb(createBreadcrumb(eventObject));
+    }
+  }
+
+  /**
+   * Creates {@link SentryEvent} from Log4j2 {@link LogEvent}.
+   *
+   * @param loggingEvent the log4j2 event
+   * @return the sentry event
+   */
+  @SuppressWarnings("JdkObsolete")
+  final @NotNull SentryEvent createEvent(@NotNull LogEvent loggingEvent) {
+    final SentryEvent event =
+        new SentryEvent(DateUtils.getDateTime(new Date(loggingEvent.getTimeMillis())));
+    final Message message = new Message();
+    message.setMessage(loggingEvent.getMessage().getFormat());
+    message.setFormatted(loggingEvent.getMessage().getFormattedMessage());
+    message.setParams(toParams(loggingEvent.getMessage().getParameters()));
+    event.setMessage(message);
+    event.setLogger(loggingEvent.getLoggerName());
+    event.setLevel(formatLevel(loggingEvent.getLevel()));
+
+    final ThrowableProxy throwableInformation = loggingEvent.getThrownProxy();
+    if (throwableInformation != null) {
+      event.setThrowable(throwableInformation.getThrowable());
+    }
+
+    if (loggingEvent.getThreadName() != null) {
+      event.setExtra("thread_name", loggingEvent.getThreadName());
+    }
+
+    final Map<String, String> mdcProperties =
+        CollectionUtils.shallowCopy(loggingEvent.getContextData().toMap());
+    if (!mdcProperties.isEmpty()) {
+      event.getContexts().put("MDC", mdcProperties);
+    }
+
+    return event;
+  }
+
+  private @NotNull List<String> toParams(@Nullable Object[] arguments) {
+    if (arguments != null) {
+      return Arrays.stream(arguments)
+          .filter(Objects::nonNull)
+          .map(Object::toString)
+          .collect(Collectors.toList());
+    } else {
+      return Collections.emptyList();
+    }
+  }
+
+  /**
+   * Creates {@link Breadcrumb} from log4j2 {@link LogEvent}.
+   *
+   * @param loggingEvent the log4j2 event
+   * @return the sentry breadcrumb
+   */
+  private @NotNull Breadcrumb createBreadcrumb(final @NotNull LogEvent loggingEvent) {
+    final Breadcrumb breadcrumb = new Breadcrumb();
+    breadcrumb.setLevel(formatLevel(loggingEvent.getLevel()));
+    breadcrumb.setCategory(loggingEvent.getLoggerName());
+    breadcrumb.setMessage(loggingEvent.getMessage().getFormattedMessage());
+    return breadcrumb;
+  }
+
+  /**
+   * Transforms a {@link Level} into an {@link SentryLevel}.
+   *
+   * @param level original level as defined in log4j.
+   * @return log level used within sentry.
+   */
+  private static @NotNull SentryLevel formatLevel(@NotNull Level level) {
+    if (level.isMoreSpecificThan(Level.ERROR)) {
```

Comment:
It doesn't have Fatal?

Reply:
no

Reply:
Good catch, Log4j2 actually does have `fatal`.

Reply:
true, i've mistaken with logback